File: ruby/dev/net/sample.rb
require "net/http"

uri = URI.parse("https://example.com/")
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = (uri.scheme == "https")

req = Net::HTTP::Get.new("/index.html")
res = http.request(req)
puts res.body
puts("---")

p http.class # Net::HTTP
p res.class # Net::HTTPOK
p res.header # #<Net::HTTPOK 200 OK readbody=true>
res.each_key { p it }
p res["Last-Modified"]

p Net::HTTP.get("example.com", "/index.html", 80) # => String
p Net::HTTP.get(uri) # => String

p Net::HTTP.get_response(uri, { "Accept" => "text/html" }) # => Net::HTTPOK

uri = URI("https://httpbin.org/post")
data = '{ "foo": "bar" }'
p Net::HTTP.post(uri, data, { "Content-Type": "application/json" }) # => #<Net::HTTPOK 200 OK readbody=true>

File: ruby/dev/net/field_work/go/sample.go
// https://pkg.go.dev/net/http?utm_source=chatgpt.com

package main

import (
	"net/http"
	"io"
	"fmt"
)

func main() {
	req, err := http.NewRequest("GET", "https://example.com", nil)

	if err != nil {
		panic(err)
	}

	req.Header.Set("Accept", "text/html")

	res, err := http.DefaultClient.Do(req)

	if err != nil {
		panic(err)
	}

	defer res.Body.Close()

	body, _ := io.ReadAll(res.Body)
	fmt.Println("GET status:", res.Status)
	fmt.Println("GET body:", string(body))

	res, err = http.Get("https://example.com")

	if err != nil {
		panic(err)
	}

	body, _ = io.ReadAll(res.Body)
	fmt.Println("GET status:", res.Status)
	fmt.Println("GET body:", string(body))
}
